From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Wed, 21 Jun 2023 09:27:28 +0200
Subject: [PATCH] http_client: fix depecration of CURLOPT_PROTOCOLS and
 CURLINFO_SIZE_DOWNLOAD

- GH #3484

(cherry picked from commit b7b3c67fc1205d114fadf360a594930ef69835a3)
---
 src/modules/http_client/functions.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/modules/http_client/functions.c b/src/modules/http_client/functions.c
index 8a2b993..be0b202 100644
--- a/src/modules/http_client/functions.c
+++ b/src/modules/http_client/functions.c
@@ -155,8 +155,12 @@ static int curL_request_url(struct sip_msg *_m, const char *_met,
 	res = curl_easy_setopt(curl, CURLOPT_URL, _url);
 
 	/* Limit to HTTP and HTTPS protocols */
+#if defined(CURL_AT_LEAST_VERSION) && CURL_AT_LEAST_VERSION(7, 85, 0)
+	res = curl_easy_setopt(curl, CURLOPT_PROTOCOLS_STR, "http,https");
+#else
 	res = curl_easy_setopt(
 			curl, CURLOPT_PROTOCOLS, CURLPROTO_HTTP | CURLPROTO_HTTPS);
+#endif
 	res = curl_easy_setopt(
 			curl, CURLOPT_REDIR_PROTOCOLS, CURLPROTO_HTTP | CURLPROTO_HTTPS);
 
@@ -383,8 +387,13 @@ static int curL_request_url(struct sip_msg *_m, const char *_met,
 
 	if((stat >= 200) && (stat < 500)) {
 		double datasize = 0;
-
+#if defined(CURL_AT_LEAST_VERSION) && CURL_AT_LEAST_VERSION(7, 55, 0)
+		curl_off_t dlsize;
+		curl_easy_getinfo(curl, CURLINFO_SIZE_DOWNLOAD_T, &dlsize);
+		download_size = (double)dlsize;
+#else
 		curl_easy_getinfo(curl, CURLINFO_SIZE_DOWNLOAD, &download_size);
+#endif
 		LM_DBG("  -- curl download size: %u \n", (unsigned int)download_size);
 		datasize = download_size;
 
